demo/src/App.js

```diff
@@ -1,15 +1,36 @@
-import React from 'react';
-import {Navbar, Nav, Image} from 'react-bootstrap';
+import React, {useMemo} from 'react';
+import {connect} from 'react-redux';
+import PropTypes from 'prop-types';
+import {Navbar, Nav, Image, Badge} from 'react-bootstrap';
 import {Link} from 'react-router-dom';
 import {Router} from './Router.js';
 import Logo from './logo.svg';
 import './App.css';
 
 /**
- * @return {!JSX} Wrapper component that wraps the entire site with
- *     common components such as a header, navigation, and shopping cart.
+ * Computes the total number of items in the cart.
+ * @param {ItemStore} items A {@link ItemStore} listing of items
+ * to purchase and their desired quantity.
+ * @return {number} The total quantity.
  */
-export function App() {
+function computeNumberOfItemsInCart(items) {
+  let totalItems = 0;
+  for (const item of Object.values(items)) {
+    totalItems += item.quantity;
+  }
+  return totalItems;
+}
+
+/**
+ * Creates wrapper component that wraps the entire site with common components
+ * such as a header, navigation, and shopping cart.
+ * @param {!ItemStore} items The items stored in the site state
+ * @return {!JSX}
+ */
+const AppBase = function({items}) {
+  const itemsInCart = useMemo(() => {
+    return computeNumberOfItemsInCart(items);
+  }, [items]);
```

Comment:
Just want to confirm due to my lack of familiarity with react.

The useMemo function here is saying that the first time the page is rendered, compute the number of items in the cart.  Then on each subsequent render, only recompute this value if the items attribute of the App has changed, i.e. the contents of the cart is different.

Is that about correct?

Reply:
Yes, that is correct :)